include/amici/event.h

```diff
@@ -96,7 +116,13 @@ class EventQueue {
             throw std::runtime_error("No pending events");
         }
 
-        // TODO: FIFO for now. Implement priority handling here.
+        // Sort events by priority from high to low.
+        pending_events_.sort([](Event const& a, Event const& b) {
+            // The priority is NaN if not defined. In this case, the execution
+            // order is undefined, so this does not need any special treatment.
+            return a.get_priority() > b.get_priority();
```

Comment:
what about mixed defined and undefined event priories (sbml spec says we can do whathever we want with undefined prios). Also in the case of equal priority SBML spec says events should be picked randomly.

Reply:
> what about mixed defined and undefined event priories (sbml spec says we can do whathever we want with undefined prios). 

That is already the case here.

> Also in the case of equal priority SBML spec says events should be picked randomly.

Ah, you are right. This will probably not work well with gradient-based optimization, but if SBML demands it, let's do it. It's the user's problem then.